transmogrifier/sources/datacite.py

```diff
@@ -117,6 +155,43 @@ def create_from_datacite_xml(
             kwargs["dates"] = [
                 Date(kind="Publication date", value=publication_year.string)
             ]
+        dates = xml.metadata.find_all("date")
+        for date in dates:
+            d = Date(value=date.string)
+            if "dateType" in date.attrs:
+                d.kind = date.attrs["dateType"]
+            if "dateInformation" in date.attrs:
+                d.note = date.attrs["dateInformation"]
+            kwargs.setdefault("dates", []).append(d)
+
+        # edition
+        edition = xml.metadata.find("version")
+        if edition:
+            kwargs["edition"] = edition.string
+
+        # file_formats
+        file_formats = xml.metadata.find_all("format")
+        for file_format in file_formats:
+            kwargs.setdefault("file_formats", []).append(file_format.string)
+
+        # funding_information
+        funding_references = xml.metadata.find_all("fundingReference")
+        for funding_reference in funding_references:
+            f = Funder(
+                funder_name=funding_reference.find("funderName").string,
+            )
+            if funding_reference.find("funderIdentifier"):
+                funder_identifier = funding_reference.find("funderIdentifier")
+                f.funder_identifier = funder_identifier.string
+                if "funderIdentifierType" in funder_identifier.attrs:
+                    f_id_type = funder_identifier.attrs["funderIdentifierType"]
+                    f.funder_identifier_type = f_id_type
+            if funding_reference.find("awardNumber"):
+                award_number = funding_reference.find("awardNumber")
+                f.award_number = award_number.string
+                if "awardURI" in award_number.attrs:
+                    f.award_uri = award_number.attrs["awardURI"]
+            kwargs.setdefault("funding_information", []).append(f)
 
         # identifiers
         identifier_xml = xml.metadata.find("identifier")
```

Comment:
Taking a 2nd look at some of the code from the previous PR, is there potentially an issue in grabbing `xml.metadata.find("identifier")` here and grabbing `xml.header.find("identifier")` for `source_record_id`?